runtime/Java/src/org/antlr/v4/runtime/tree/Trees.java

```diff
@@ -275,6 +289,47 @@ public static ParserRuleContext getRootOfSubtreeEnclosingRegion(ParseTree t,
 		return null;
 	}
 
+	/** Replace any subtree siblings of root that are completely to left
+	 *  or right of lookahead range with a CommonToken(Token.INVALID_TYPE,"...")
+	 *  node. The source interval for t is not altered to suit smaller range!
+	 *
+	 *  WARNING: destructive to t.
+	 *
+	 *  @since 4.5.1
+	 */
+	public static void stripChildrenOutOfRange(ParserRuleContext t,
+											   ParserRuleContext root,
+											   int startIndex,
+											   int stopIndex)
+	{
+		if ( t==null ) return;
+		for (int i = 0; i < t.getChildCount(); i++) {
+			ParseTree child = t.getChild(i);
+			Interval range = child.getSourceInterval();
+			if ( child instanceof ParserRuleContext && (range.b < startIndex || range.a > stopIndex) ) {
+				if ( isAncestorOf(child, root) ) { // replace only if subtree doesn't have displayed root
+					CommonToken abbrev = new CommonToken(Token.INVALID_TYPE, "...");
+					t.children.set(i, new TerminalNodeImpl(abbrev));
+				}
+			}
+		}
+	}
+
+	/** Return first node satisfying the pred
+	 *
+ 	 *  @since 4.5.1
+	 */
+	public static Tree findNodeSuchThat(Tree t, Predicate<Tree> pred) {
```

Comment:
A `Predicate<T>` takes an argument `T` and returns a `boolean`. A `Predicate<Object>` clearly knows how to handle a `Tree` because a `Tree` is an `Object`. However, with the current definition you couldn't use the following:

```
static final Predicate<Object> TRUE = new Predicate<Object>() {
  @Override
  public boolean test(Object o) {
    return true;
  }
};
```

The type which means "A predicate which understands a `Tree` argument" is `Predicate<? super Tree>`.